solana/pyth2wormhole/client/src/main.rs

```diff
@@ -84,63 +98,63 @@ fn main() -> Result<(), ErrBox> {
 
     let (recent_blockhash, _) = rpc_client.get_recent_blockhash()?;
 
-    let tx = match cli.action {
+    let txs = match cli.action {
         Action::Init {
             owner_addr,
             pyth_owner_addr,
             wh_prog,
-        } => handle_init(
+        } => vec![handle_init(
             payer,
             p2w_addr,
             owner_addr,
             wh_prog,
             pyth_owner_addr,
             recent_blockhash,
-        )?,
+        )?],
         Action::SetConfig {
             ref owner,
             new_owner_addr,
             new_wh_prog,
             new_pyth_owner_addr,
-        } => handle_set_config(
+        } => vec![handle_set_config(
             payer,
             p2w_addr,
             read_keypair_file(&*shellexpand::tilde(&owner))?,
             new_owner_addr,
             new_wh_prog,
             new_pyth_owner_addr,
             recent_blockhash,
-        )?,
+        )?],
         Action::Attest {
-            product_addr,
-            price_addr,
-            nonce,
+            ref attestation_cfg,
+            start_nonce,
         } => handle_attest(
             &rpc_client,
             payer,
             p2w_addr,
-            product_addr,
-            price_addr,
-            nonce,
+            attestation_cfg.as_path(),
+            start_nonce,
```

Comment:
how do these nonces work? do we lose the state of the nonce when this program quits? Who's responsible for maintaining what the last used nonce is?

Reply:
I am not sure. Wormhole accepts a nonce in its payload struct (PostMessageData). however It does not do anything significant with it (see `solana/bridge/program/src/api/post_message.rs`) in its contract. We might be well off without it.

Reply:
The nonce is redundant and I removed it